core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -682,266 +706,166 @@ private void syncRemoteToLocal() throws InterruptedException {
+              if (lastSeenHasNoVersion) {
+                action = SyncAction.APPLY_AND_VERSION_FROM_REMOTE;
+              } else {
+                action = lastSeenHash != remoteHash
+                    ? SyncAction.APPLY_FROM_REMOTE : SyncAction.DROP;
+              }
+            } else if (lastSeenHasNoVersion) {
+              action = lastSeenHash != remoteHash ? SyncAction.APPLY_FROM_REMOTE : SyncAction.DROP;
+              message = SyncMessage.EMPTY_VERSION_MESSAGE;
+            } else {
+              final long remoteVersionCounter = remoteVersion.getVersionCounter();
+              final long lastSeenVersionCounter = lastSeenVersion.getVersionCounter();
 
-    final DocumentVersionInfo newestRemoteVersionInfo;
-    try {
-      newestRemoteVersionInfo = DocumentVersionInfo
-          .getRemoteVersionInfo(newestRemoteDocument);
-    } catch (final Exception e) {
-      emitError(docConfig,
-          String.format(
-              Locale.US,
-              "t='%d': syncRemoteChangeEventToLocal ns=%s documentId=%s got a remote "
-                  + "document that could not have its version info parsed "
-                  + "; dropping the event, and desyncing the document",
-              logicalT,
-              nsConfig.getNamespace(),
-              docConfig.getDocumentId()));
-      return desyncDocumentsFromRemote(nsConfig.getNamespace(), docConfig.getDocumentId());
-    }
-
-    // ii. If the current GUID of the remote document (as determined by this lookup) is equal
-    //     to the GUID of the local document, drop the event. We're believed to be behind in
-    //     the change stream at this point.
-    if (newestRemoteVersionInfo.hasVersion()
-            && newestRemoteVersionInfo.getVersion().getInstanceId()
-                    .equals(localVersion.instanceId)) {
-
-      logger.debug(String.format(
-              Locale.US,
-              "t='%d': syncRemoteChangeEventToLocal ns=%s documentId=%s latest document lookup "
-                      + "indicates that this is a stale event; dropping the event",
-              logicalT,
-              nsConfig.getNamespace(),
-              docConfig.getDocumentId()));
-      return null;
+              if (remoteVersionCounter > lastSeenVersionCounter) {
+                action = SyncAction.APPLY_FROM_REMOTE;
+                message = SyncMessage.APPLY_FROM_REMOTE_MESSAGE;
+              } else {
+                if (lastSeenVersion.getSyncProtocolVersion() == SYNC_PROTOCOL_VERSION
+                    && lastSeenVersion.getInstanceId().equals(remoteVersion.getInstanceId())) {
+                  message = SyncMessage.PROBABLY_GENERATED_BY_US_MESSAGE;
+                  action = SyncAction.DROP;
+                } else if (remoteVersionCounter == lastSeenVersionCounter
+                    && lastSeenHash != remoteHash) {
+                  message = SyncMessage.REMOTE_UPDATE_WITHOUT_VERSION_MESSAGE;
+                  action = SyncAction.SYNTHETIC_CONFLICT;
```

Comment:
this handles the case of an external update from outside of sync (eg from web or a server application) where the value was changed but not the sync version. the conflict forces the user-specified logic to decide what to do with this (e.g. remote wins by default) -- hmm I guess it makes more sense for this to apply if different rather than conflict...